controller/src/main/java/org/jboss/as/controller/OperationContextImpl.java

```diff
@@ -867,6 +869,8 @@ private void ensureWriteLockForRuntime() {
                     // different messages for both so just throwing a RuntimeException to get the automatic handling
                     // in AbstractOperationContext.executeStep is not what I wanted
                     ControllerLogger.MGMT_OP_LOGGER.timeoutAwaitingInitialStability(timeout / 1000, activeStep.operationId.name, activeStep.operationId.address);
+                    // Produce and log thread dump for diagnostics
+                    ThreadDumpUtil.threadDump();
```

Comment:
@aogburn Would you mind moving the call after the `setRollbackOnly();`?

I could be too conservative, but if there is an error propagated from the `ThreadDumpUtil` it could be better to still ensure the `setRollbackOnly();` was called.